brave/brave-core
# feat(wallet): Swap and Bridge Route Selection Modal

## Description 

- Introduces a `Route Selection` modal that allows users to change routes when creating a `Swap` or `Bridge` transaction.
- Changes the `New Quote Countdown` time from `10` seconds to `30` seconds


Resolves <https://github.com/brave/brave-browser/issues/38648>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Open the `Wallet` and navigate to the `Swap` page
2. Select some tokens to swap and get a `quote`
3. After you receive your `quote`, expand the `Advanced` section of the `Quote Info`
4. Open the `Route Selection` modal and view the available routes.
5. Test changing `Routes` and make sure all data is accurate to your selection
6. Follow these same steps for `Bridge`

Swap:

https://github.com/user-attachments/assets/44aa6bf8-9b43-4ffb-a9ef-0171f402abc7

Bridge:

https://github.com/user-attachments/assets/771d8d38-1475-4547-84c7-e6f59f75ca57
